google/cloud/pubsub/internal/tracing_message_batch.cc

```diff
@@ -33,21 +33,36 @@ void TracingMessageBatch::SaveMessage(pubsub::Message m) {
   auto active_span = opentelemetry::trace::GetSpan(
       opentelemetry::context::RuntimeContext::GetCurrent());
   active_span->AddEvent("gl-cpp.added_to_batch");
-  message_spans_.push_back(active_span);
+  {
+    std::lock_guard<std::mutex> lk(mu_);
+    message_spans_.push_back(active_span);
```

Comment:
I am not sure, but maybe `std::move(active_span)`?

Reply:
Done.